### PR TITLE
Refactor `Variable.zerograd` test 

### PR DESCRIPTION
**Pull request by niboshi**
Separated the test and fixed to use `inject_backend_tests`.

~Depends on #7191~

### CODE REVIEW

File: tests/chainer_tests/test_variable.py [hunk 1]
```diff
@@ -831,6 +751,46 @@ def test_grad_var(self, backend_config):
...
+    {
+        'shape': [(10,), (0,), ()],
+        'grad_var_requires_grad': [True, False],
+        'fill': [True, False],
+    }
+))
+class TestVariableZerogad(unittest.TestCase):
+
+    def test_zerograd(self, backend_config):
+        shape = (3,)
```
Comment (asi1024) [inline tests/chainer_tests/test_variable.py]:
```suggestion
        shape = self.shape
```

Comment (niboshi) [reply tests/chainer_tests/test_variable.py]: Fixed.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
Jenkins failure seems unrelated to this PR. LGTM.

**Pull request merged**